samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -0,0 +1,144 @@
+                LOG.warning("FunctionName is defined using the value provided for --resource-id option.")
+            elif parameter_key == "Payload":
+                LOG.warning(
+                    "Payload is defined using the value provided for either --payload or --payload-file options."
+                )
+            else:
+                self.request_parameters[parameter_key] = parameter_value
+
+    def _execute_action(self, payload: str):
+        self.request_parameters["FunctionName"] = self._function_name
+        self.request_parameters["Payload"] = payload
+        LOG.debug(
+            "Calling lambda_client.invoke with FunctionName:%s, Payload:%s, parameters:%s",
+            self._function_name,
+            payload,
+            self.request_parameters,
+        )
+        try:
+            response = self._lambda_client.invoke(**self.request_parameters)
+        except ParamValidationError as param_val_ex:
+            raise InvalidResourceBotoParameterException(
+                f"Invalid parameter key provided."
+                f" {str(param_val_ex).replace('FunctionName, ', '').replace('Payload, ', '')}"
+            ) from param_val_ex
+        except ClientError as client_ex:
+            validation_err_str = "ValidationException"
+            if boto_utils.get_client_error_code(client_ex) == validation_err_str:
+                raise InvalidResourceBotoParameterException(
+                    f"Invalid parameter value provided. {str(client_ex).replace(f'({validation_err_str}) ', '')}"
+                ) from client_ex
+            raise client_ex
+        return response
+
+
+class DefaultConvertToJSON(RemoteInvokeRequestResponseMapper):
+    """
+    If a regular string is provided as payload, this class will convert it into a JSON object
+    """
+
+    def map(self, test_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
+        if not test_input.is_file_provided():
+            try:
+                _ = json.loads(cast(str, test_input.payload))
+            except JSONDecodeError:
+                json_value = json.dumps(test_input.payload)
+                LOG.info(
+                    "Auto converting value '%s' into JSON '%s'. "
+                    "If you don't want auto-conversion, please provide a JSON string as payload",
+                    test_input.payload,
+                    json_value,
+                )
+                test_input.payload = json_value
+
+        return test_input
+
+
+class LambdaResponseConverter(RemoteInvokeRequestResponseMapper):
+    """
+    This class helps to convert response from lambda service. Normally lambda service
+    returns 'Payload' field as stream, this class converts that stream into string object
+    """
+
+    def map(self, remote_invoke_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
+        assert isinstance(remote_invoke_input.response, dict), "Invalid response type received, expecting dict"
```

Comment:
Not sure how `assert` keyword will behave here, can we instead verify with something else and raise an exception?

Reply:
Addressed in the new commit.